lib/vanagon/platform.rb

```diff
@@ -436,6 +450,18 @@ def is_cross_compiled_linux?
       return (is_cross_compiled? && is_linux?)
     end
 
+    # Utility matcher to determine whether the platform uses the system provided toolchain
+    # as opposed to pl-build-tools.
+    #
+    # @return [true, false] true if it is uses the system toolchain, false otherwise
+    def uses_system_toolchain? # rubocop:disable Metrics/PerceivedComplexity
```

Comment:
My only hesitation with this is that it feels pretty Puppet-specific, whereas vanagon _theoretically_ should be more general than that. 
I do, however, understand the desire to not repeat this code across several projects. I guess I'm wondering if there's a different shared place for these settings? like maybe a shared puppet-runtime setting?
But if we already have a bunch of stuff like this in vanagon, I guess I won't stop y'all from adding more.. 
Just something to consider..

Reply:
Agree with this. Maybe we can find a better way around this.

Reply:
@mwaggett fair enough, I'll remove this method and we'll reverse the logic in puppet-runtime and agent instead.

I'll repurpose this PR to only have the Debian/Ubuntu OS matchers.